association-frontend/src/components/button/index.tsx

```diff
@@ -1,40 +1,59 @@
 import classnames from "classnames";
-import React from "react";
+import React, { useEffect, useState } from "react";
 
 type ButtonProps = {
   disabled?: boolean;
   text?: string;
   link?: string;
   type?: "button" | "reset" | "submit";
   fullWidth?: boolean;
+  loading?: boolean;
   onClick?: (...args: any[]) => void;
 };
 
+const loadingEmojis = ["🐍", "🕑", "🕓", "🕗", "🕙", "🧨"];
+
 export const Button: React.FC<ButtonProps> = ({
   type,
   link,
   text,
   fullWidth,
   children,
+  loading = false,
   ...props
 }) => {
+  const [emojiIndex, setEmojiIndex] = useState(0);
+
+  useEffect(() => {
+    if (loading) {
+      const interval = setInterval(() => {
+        setEmojiIndex((emojiIndex + 1) % loadingEmojis.length);
+      }, 600);
+      return () => clearInterval(interval);
+    }
+  });
```

Comment:
you are missing the deps here meaning that this effect is called at every render, it seems that we want this only when `loading` changes?

so 

```suggestion
  }, [loading]);
```